code/modules/projectiles/updated_projectiles/ammunition.dm

```diff
@@ -355,6 +355,165 @@ Turn() or Shift() as there is virtually no overhead. ~N
+		to_chat(user, "<span class='warning'>The [src] is already full.")
+		return
+	var/obj/item/ammo_magazine/MG = I
+	if(MG.default_ammo != ammo_type || MG.max_rounds != max_magazine_rounds)
+		to_chat(user, "<span class='warning'>That's not the right kind of ammo.</span>")
+		return
+	del user.get_held_item()
+	magazine_amount += 1
+	update_icon()
+
+
+/obj/item/ammobox/attack_hand(mob/user)
+	if(deployed == 0)
+		var/obj/item/ammobox/I = new base
+		I.magazine_amount = magazine_amount
+		user.put_in_hands(I)
+		cdel(src)
+		return
+	if(magazine_amount == 0)
+		to_chat(user, "<span class='warning'>The [src] is empty.")
+		return
+	var/N = new magazine_type
+	user.put_in_hands(N)
+	magazine_amount -= 1
+	update_icon()
+
+
+/obj/item/ammobox/attack_self(mob/user)
+	deployed = 1
+	update_icon()
+	user.drop_held_item(src)
+
+
+/obj/item/ammobox/MouseDrop(atom/over_object)
+	if(deployed == 0)
+		return
+	if(istype(over_object, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = over_object
+		if(H==usr && !H.is_mob_incapacitated() && Adjacent(H) && in_range(src, over_object))
+			var/obj/item/ammobox/I = new base
+			I.magazine_amount = magazine_amount
+			H.put_in_hands(I)
+			icon_state = base_icon_state
+			cdel(src)
+
+
+//Deployable shotgun ammo box
+
+/obj/item/ammo_magazine/shotgunbox
+	name = "Slug Ammo Box"
+	desc = "A large, deployable ammo box."
+	icon = 'icons/obj/items/ammo.dmi'
+	icon_state = "ammoboxslug"
+	item_state = "ammoboxslug"
+	var/base_icon_state = "ammoboxslug"
+	default_ammo = /datum/ammo/bullet/shotgun/slug
+	caliber = "12g"
+	gun_type = /obj/item/weapon/gun/shotgun
+	max_rounds = 100
+	current_rounds = 100
+	w_class = 5
+	var/base = /obj/item/ammo_magazine/shotgunbox
+	var/deployed = 0
+
+
+/obj/item/ammo_magazine/shotgunbox/update_icon()
+	if(current_rounds > 0) 
+		icon_state = "[base_icon_state]_deployed"
+	else 
+		icon_state = "[base_icon_state]_empty"
+
+/obj/item/ammo_magazine/shotgunbox/attack_self(mob/user)
+	deployed = 1
+	update_icon()
+	user.drop_held_item(src)
+
+/obj/item/ammo_magazine/shotgunbox/MouseDrop(atom/over_object)
+	if(deployed == 0)
+		return
+	if(istype(over_object, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = over_object
+		if(H==usr && !H.is_mob_incapacitated() && Adjacent(H) && in_range(src, over_object))
+			var/obj/item/ammo_magazine/shotgunbox/I = new base
+			I.current_rounds = current_rounds
+			H.put_in_hands(I)
+			icon_state = base_icon_state
+			cdel(src)
+
+/obj/item/ammo_magazine/shotgunbox/examine(mob/user)
+	to_chat(user, "It contains [current_rounds] out of [max_rounds] rounds.")
+
+/obj/item/ammo_magazine/shotgunbox/attack_hand(mob/user)
+	if(deployed == 0)
+		var/obj/item/ammo_magazine/shotgunbox/I = new base
+		I.current_rounds = current_rounds
+		user.put_in_hands(I)
+		cdel(src)
+		return
+	if(flags_magazine & AMMUNITION_REFILLABLE)
+		if (current_rounds > 0)
+			if(create_handful(user))
+				update_icon()
+				return
+			else
```

Comment:
Unless if you want to combine it with this `else` for a proper message answer and earlier return.